File: LeetCode/Java/UniquePaths.java
import java.util.Arrays;

public class UniquePaths {


public int uniquePaths(int m, int n) {
    int[] aboveRow = new int[n];
    Arrays.fill(aboveRow, 1);

    for (int row = 1; row < m; row++) {
        int[] currentRow = new int[n];
        Arrays.fill(currentRow, 1);

        for (int col = 1; col < n; col++) {
            currentRow[col] = currentRow[col - 1] + aboveRow[col];
        }

        aboveRow = currentRow;
    }

    return aboveRow[n - 1];
}

public static void main(String[] args) {
    UniquePaths up = new UniquePaths();
    int m=3,n=7;
    System.out.println("Unique Paths: "+up.uniquePaths(m,n));
    }
}

File: LeetCode/Python/UniquePaths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        
        aboveRow = [1] * n

        for _ in range(m - 1):
            currentRow = [1] * n
            for i in range(1, n):
                currentRow[i] = currentRow[i-1] + aboveRow[i]
            aboveRow = currentRow
        
        return aboveRow[-1]
    
s=Solution()

m = 3
n = 7

print("Number of possible unique paths: ",s.uniquePaths(m,n))


# Test Cases

# Input: m = 3, n = 7
# Output: 28

# Input: m = 3, n = 2
# Output: 3